### PR TITLE
qa/tasks: Fix raises that doesn't re-raise

### PR DESCRIPTION
**Pull request by joscollin**
Based on the discussion: https://github.com/ceph/ceph/pull/30058#discussion_r321749572

Fixes: https://tracker.ceph.com/issues/41694

### CODE REVIEW

File: qa/tasks/cephfs/test_volumes.py [hunk 1]
```diff
@@ -261,7 +262,8 @@ def test_subvolume_group_create_with_invalid_data_pool_layout(self):
             if ce.exitstatus != errno.EINVAL:
                 raise
         else:
-            raise
+            raise RuntimeError("expected the 'fs subvolume create' command to fail")
```
Comment (ajarr) [inline qa/tasks/cephfs/test_volumes.py]: s/subvolume/subvolumegroup/

File: qa/tasks/cephfs/test_volumes.py [hunk 2]
```diff
@@ -279,7 +281,7 @@ def test_subvolume_group_create_with_auto_cleanup_on_fail(self):
             if ce.exitstatus != errno.ENOENT:
                 raise
         else:
-            raise
+            raise RuntimeError("expected the 'fs subvolume getpath' command to fail")
```
Comment (ajarr) [inline qa/tasks/cephfs/test_volumes.py]: s/subvolume/subvolumegroup/

**Comment by ajarr** [comment]
@joscollin as part of this PR, but a different commit, can you also raise exceptions when the test expects the command to fail, but the command succeeds and the test still passes? I think there are a couple of instances of that in this test module.

**Comment by joscollin** [comment]
> @joscollin as part of this PR, but a different commit, can you also raise exceptions when the test expects the command to fail, but the command succeeds and the test still passes? I think there are a couple of instances of that in this test module.

I think these changes would create unreachable code.
For instance, see [this line](https://github.com/ceph/ceph/pull/30201/files#diff-466824cf9c21788bb09544d23f2bce87R148). What do you suggest in that case? Keep the unreachable line inside `else:` ?

**Comment by joscollin** [comment]
> @joscollin as part of this PR, but a different commit, can you also raise exceptions when the test expects the command to fail, but the command succeeds and the test still passes? I think there are a couple of instances of that in this test module.

Checking it.

**Comment by joscollin** [comment]
@ajarr Done. Please check again.

**Comment by joscollin** [comment]
jenkins test make check

**Comment by joscollin** [comment]
jenkins test make check

File: qa/tasks/cephfs/test_volumes.py [hunk 3]
```diff
@@ -114,6 +114,7 @@ def test_subvolume_create_and_rm(self):
         except CommandFailedError as ce:
             if ce.exitstatus != errno.ENOENT:
                 raise
+        raise RuntimeError("expected the 'fs subvolume getpath' command to fail. Subvolume not removed.")
```
Comment (ajarr) [inline qa/tasks/cephfs/test_volumes.py]: This needs to be within `else` clause

Comment (joscollin) [reply qa/tasks/cephfs/test_volumes.py]: Good catch. Fixed.

**Comment by joscollin** [comment]
jenkins test make check

File: qa/tasks/cephfs/test_volumes.py [hunk 4]
```diff
@@ -142,9 +144,7 @@ def test_subvolume_create_with_invalid_data_pool_layout(self):
             if ce.exitstatus != errno.EINVAL:
                 raise
         else:
-            raise
-        # clean up
-        self._fs_cmd("subvolume", "rm", self.volname, subvolume, "--force")
```
Comment (ajarr) [inline qa/tasks/cephfs/test_volumes.py]: As you're removing this, you'll need to write a test that `fs subvolume rm --force` works as expected.

File: qa/tasks/cephfs/test_volumes.py [hunk 5]
```diff
@@ -261,9 +262,7 @@ def test_subvolume_group_create_with_invalid_data_pool_layout(self):
             if ce.exitstatus != errno.EINVAL:
                 raise
         else:
-            raise
-        # clean up
-        self._fs_cmd("subvolumegroup", "rm", self.volname, group, "--force")
```
Comment (ajarr) [inline qa/tasks/cephfs/test_volumes.py]: As you're removing this, you'll need to write a separate test that `fs subvolumegroup rm --force` works as expected.

File: qa/tasks/cephfs/test_volumes.py [hunk 6]
```diff
@@ -465,9 +463,8 @@ def test_nonexistent_subvolume_snapshot_rm(self):
         except CommandFailedError as ce:
             if ce.exitstatus != errno.ENOENT:
                 raise
-
-        # force remove snapshot
-        self._fs_cmd("subvolume", "snapshot", "rm", self.volname, subvolume, snapshot, "--force")
```
Comment (ajarr) [inline qa/tasks/cephfs/test_volumes.py]: As you're removing this, you'll need to write a separate test that `fs subvolume snapshot rm --force` works as expected.

**Comment by ajarr** [comment]
And maybe add a small test to check whether `fs subvolumegroup snapshot rm --force` works as expected?

**Comment by joscollin** [comment]
@ajarr ping

**Comment by joscollin** [comment]
jenkins test make check

**Comment by joscollin** [comment]
jenkins test make check

**Comment by joscollin** [comment]
retest this please

**Comment by joscollin** [comment]
retest this please

**Comment by joscollin** [comment]
http://pulpito.front.sepia.ceph.com/jcollin-2019-12-05_14:20:19-fs-wip-pr-30201-31961-testing3-distro-basic-smithi/

**Pull request merged**